debezium-embedded/src/test/java/io/debezium/embedded/EmbeddedEngineTest.java

```diff
@@ -260,6 +263,54 @@ public void interruptedOffsetCommitShutsDown() throws Exception {
         assertThat(firstLatch.getCount()).isEqualTo(0);
     }
 
+    @Test
+    public void shouldRunSpecifiedTaskConfiguration() throws Exception {
+        var offset = 10_000;
+
+        Configuration config = Configuration.create()
+                .with(EmbeddedEngine.ENGINE_NAME, "debezium-engine")
+                .with(EmbeddedEngine.CONNECTOR_CLASS, SimpleSourceConnector.class.getName())
+                .with(EmbeddedEngine.MAX_TASKS, 3)
+                .with(EmbeddedEngine.TASK_ID, 1)
+                .with(StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG, OFFSET_STORE_PATH)
+                .with(SimpleSourceConnector.BATCH_COUNT, 2)
+                .with(SimpleSourceConnector.RECORD_COUNT_PER_BATCH, 5)
+                .with(SimpleSourceConnector.TASK_VALUE_OFFSET, offset)
+                .build();
+
+        CountDownLatch latch = new CountDownLatch(2 * 5);
+
+        var actual = new HashSet<Integer>();
+        var expected = IntStream.range(offset + 1, offset + 2 * 5 + 1).boxed().toArray();
+
+        DebeziumEngine.Builder<ChangeEvent<SourceRecord, SourceRecord>> engineBuilder = DebeziumEngine.create(Connect.class, Connect.class)
+                .using(config.asProperties())
+                .notifying((events, committer) -> {
+                    for (var event : events) {
+                        SourceRecord record = event.value();
+                        Struct key = (Struct) record.key();
+                        int id = key.getInt32("id");
+                        actual.add(id);
+                        latch.countDown();
+                        committer.markProcessed(event);
+                    }
+                    committer.markBatchFinished();
+                })
+                .using(this.getClass().getClassLoader());
+
+        try (var engine = engineBuilder.build()) {
```

Comment:
What would be the recommended way to speedup the shutwodn? By default the engine will wait before stopping by the amount specified in ``EmbeddedEngine#WAIT_FOR_COMPLETION_BEFORE_INTERRUPT_PROP`` which is by default 5 minutes.

Reply:
The reason is that the `SimpleSourceConnector#poll()` purposefully blocks forever. This will have to be changed or made configurable.